services/dag-indexer/src/mappings/handleCall.ts

```diff
@@ -0,0 +1,77 @@
+global.TextEncoder = require('util').TextEncoder
+global.TextDecoder = require('util').TextDecoder
+global.Buffer = require('buffer/').Buffer
+
+import {
+  blake3HashFromCid,
+  cidOfNode,
+  cidToString,
+  decodeNode,
+  IPLDNodeData,
+  PBNode,
+} from '@autonomys/auto-dag-data'
+import { SubstrateExtrinsic } from '@subql/types'
+import { ExtrinsicPrimitive } from './utils'
+import { Node } from '../types/models/Node'
+
+export async function handleCall(_call: SubstrateExtrinsic): Promise<void> {
+  const {
+    idx: indexInBlock,
+    block: {
+      timestamp,
+      block: {
+        header: { number, hash: _blockHash },
+      },
+    },
+    extrinsic: { method, hash },
+    success,
+  } = _call
+  try {
+    // Skip if extrinsic failed
+    if (!success) return
+
+    const blockHeight = BigInt(number.toString())
+    const blockHash = _blockHash.toString()
+    const extrinsicId = `${blockHeight}-${indexInBlock}`
```

Comment:
Do we reset the index if there's a rollback / fork change? Or is that handled in the upstream data?

Reply:
Yes, it's handled by subquery

Reply:
Saeid made some [improvements](https://github.com/autonomys/subql/pulls?q=is%3Apr+is%3Aclosed) to how this is handled, as it was tricky with our "finalization".